source/common/upstream/load_balancer_impl.h

```diff
@@ -411,17 +430,34 @@ class RoundRobinLoadBalancer : public EdfLoadBalancerBase {
     // already exists. Note that host sources will never be removed, but given how uncommon this
     // is it probably doesn't matter.
     rr_indexes_.insert({source, seed_});
+    // If the list of hosts changes, the order of picks change. Discard the
+    // index.
+    peekahead_index_ = 0;
   }
   double hostWeight(const Host& host) override { return host.weight(); }
+  HostConstSharedPtr unweightedHostPeek(const HostVector& hosts_to_use,
+                                        const HostsSource& source) const override {
+    uint64_t* peekahead_index = const_cast<uint64_t*>(&peekahead_index_);
+    auto i = rr_indexes_.find(source);
+    if (i == rr_indexes_.end()) {
+      return nullptr;
+    }
```

Comment:
Why is this checked here but an ASSERT below? It seems like we should be consistent. If not can you add a comment?